rust/metaplex/program/src/instruction.rs

```diff
@@ -1099,6 +1120,40 @@ pub fn create_set_store_instruction(
     }
 }
 
+/// Creates an SetStore instruction
+pub fn create_set_store_v2_instruction(
+    program_id: Pubkey,
+    store: Pubkey,
+    config: Pubkey,
+    admin: Pubkey,
+    payer: Pubkey,
+    public: bool,
+    settings_uri: Option<String>,
+) -> Instruction {
+    let accounts = vec![
+        AccountMeta::new(store, false),
+        AccountMeta::new(config, false),
+        AccountMeta::new_readonly(admin, true),
+        AccountMeta::new_readonly(payer, true),
```

Comment:
payer should be writable because this account will pay for new account creation
so here should be AccountMeta::new(payer, true)